ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -30,15 +30,31 @@
 
 #include <fstream>
 
+namespace gcs = google::cloud::storage;
+
 gcp_connection::gcp_connection(const std::string &bucket_name)
     : _gcp_client(google::cloud::storage::Client()), _bucket_name(bucket_name)
 {
 }
 
 // Builds a list of object names from the bucket.
 int
-gcp_connection::list_objects(std::vector<std::string> &objects) const
+gcp_connection::list_objects(std::vector<std::string> &objects, bool list_single) const
 {
+    using namespace gcs;
+    gcs::Client client = _gcp_client;
+    std::string const bucketName = gcp_connection::_bucket_name;
+    
+    for (auto&& object_metadata : client.ListObjects(bucketName)) {
+        if (!object_metadata) throw std::move(object_metadata).status();
+
+        objects.push_back(object_metadata->name());
+        
+        if (list_single){
```

Comment:
Can we tell google cloud to only list one object from the start as a parameter? If so we can remove this.

Reply:
From looking at the optional arguments ListObjects() takes in, I don't think google cloud can directly limit the number of objects listed, so this section is kept for now.